eradiate/scenes/atmosphere/_molecules.py

```diff
@@ -0,0 +1,440 @@
+from ...units import unit_registry as ureg
+
+
+@atmosphere_factory.register(
+    type_id="molecular_atmosphere", dict_constructor="afgl1986"
+)
+@parse_docs
+@attr.s
+class MolecularAtmosphere(Atmosphere):
+    """
+    Molecular atmosphere.
+
+    .. admonition:: Class method constructors
+
+       .. autosummary::
+
+          afgl1986
+          ussa1976
+    """
+
+    _thermoprops: xr.Dataset = documented(
+        attr.ib(
+            factory=us76.make_profile,
+            validator=attr.validators.instance_of(xr.Dataset),
+        ),
+        doc="Thermophysical properties.",
+        type=":class:`~xarray.Dataset`",
+    )
+
+    phase: PhaseFunction = documented(
+        attr.ib(
+            factory=lambda: RayleighPhaseFunction(),
+            converter=phase_function_factory.convert,
+            validator=attr.validators.instance_of(PhaseFunction),
+        )
+    )
+
+    has_absorption: Optional[bool] = documented(
+        attr.ib(
+            default=True,
+            converter=bool,
+            validator=attr.validators.instance_of(bool),
+        ),
+        doc="Absorption switch. If ``True``, the absorption coefficient is "
+        "computed. Else, the absorption coefficient is not computed and "
+        "instead set to zero.",
+        type="bool",
+        default="True",
+    )
+
+    has_scattering: Optional[bool] = documented(
+        attr.ib(
+            default=True,
+            converter=bool,
+            validator=attr.validators.instance_of(bool),
+        ),
+        doc="Scattering switch. If ``True``, the scattering coefficient is "
+        "computed. Else, the scattering coefficient is not computed and "
+        "instead set to zero.",
+        type="bool",
+        default="True",
+    )
+
+    absorption_data_sets: Optional[MutableMapping[str, str]] = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(dict),
+            validator=attr.validators.optional(attr.validators.instance_of(dict)),
+        ),
+        doc="Mapping of species and absorption data set files paths. If "
+        "``None``, the default absorption data sets are used to compute "
+        "the absorption coefficient. If not ``None``, the absorption data "
+        "set files whose paths are provided in the mapping will be used to "
+        "compute the absorption coefficient. If the mapping does not "
+        "include all species from the atmospheric "
+        "thermophysical profile, the default data sets will be used to "
+        "compute the absorption coefficient of the corresponding species.",
+        type="dict",
+    )
+
+    albedo_filename: str = documented(
+        attr.ib(
+            default="albedo.vol",
+            converter=str,
+            validator=attr.validators.instance_of(str),
+        ),
+        doc="Name of the albedo volume data file.",
+        type="str",
+        default='"albedo.vol"',
+    )
```

Comment:
Side remark: I'm still thinking about how to get rid of these, and I have a design solution :)